geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -1550,12 +1574,64 @@ public void refuseHandshake(OutputStream out, String message, byte exception) th
     out.flush();
   }
 
-  private boolean handOffQueueInitialization(Socket socket, CommunicationMode communicationMode) {
+  @Override
+  public void refuseHandshake(OutputStream out, String message, byte exception,
+      NioSslEngine sslEngine, Socket socket) throws IOException {
+    if (sslEngine == null) {
+      refuseHandshake(out, message, exception);
+      return;
+    }
+    try (ByteBufferOutputStream bbos =
+        new ByteBufferOutputStream(sslEngine.getEngine().getSession().getPacketBufferSize())) {
+
+      DataOutputStream dos = new DataOutputStream(bbos);
+
+      // Write refused reply
+      dos.writeByte(exception);
+
+      // write dummy endpointType
+      dos.writeByte(0);
+      // write dummy queueSize
+      dos.writeInt(0);
+
+      // Write the server's member
+      DistributedMember member = InternalDistributedSystem.getAnyInstance().getDistributedMember();
```

Comment:
Here's another call to reach out for a dependency and it's even worse because it has to go to a singleton which we've been working to get rid of. This one might be an example of one that should be passed into the constructor since it never changes during the life of this acceptor instance:
```
private final DistributedMember member;

AcceptorImpl(..., DistributedMember member) {
  ...
  this.member = member;
}
```